tempto-examples/src/main/java/io/prestosql/tempto/examples/ResourcesTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prestosql.tempto.examples;
+
+import io.prestosql.tempto.BeforeTestWithContext;
+import io.prestosql.tempto.ProductTest;
+import org.testng.annotations.Test;
+
+import static io.prestosql.tempto.internal.fulfillment.resources.SuiteResourcesState.closeAfterSuite;
+import static io.prestosql.tempto.internal.fulfillment.resources.TestResourcesState.closeAfterTest;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ResourcesTest
+        extends ProductTest
+{
+    private boolean testResource;
+    private boolean suiteResource;
+
+    @BeforeTestWithContext
+    public void suiteResources()
+    {
+        suiteResource = true;
+        closeAfterSuite(() -> suiteResource = false);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            if (suiteResource) {
```

Comment:
if you call `closer.close()` in `@AfterTest/@AfterClass`, you can do this verification in another `@AfterTest/@AfterClass`, so shutdown hook is not needed

Reply:
I cannot. The order of `@After*` methods is not defined. So I need to verify it after all `@AfterClass` are executed.

Reply:
You can also create package-level superclass for this check. @Before/After order in superclass is predictable and documented. 

The problem with add Shutdown hook is that it won't be reported by our gha/testng plugin. The superclass approach is also ugly, but errors will be properly reported.

Reply:
Oh... I tried and it didn't work. It is not bound to `@AfterClass`, but to `io.prestosql.tempto.internal.initialization.TestInitializationListener#onFinish` which is even later.